modules/auxiliary/gather/windows_secrets_dump.rb

```diff
@@ -871,18 +903,25 @@ def dump_ntds_hashes
 
     print_line("\n# Kerberos keys:")
     user_info.each do |_sid, info|
-      full_name = "#{domain_name}\\#{info[:username]}"
+      full_name = info[:domain_name].blank? ? info[:username] : "#{info[:domain_name]}\\#{info[:username]}"
 
       if info[:kerberos_keys].nil? || info[:kerberos_keys].empty?
         vprint_line("No Kerberos keys for #{full_name}")
       else
-        credential_opts[:type] = :nonreplayable_hash
-        info[:kerberos_keys].each do |key_type, key_value|
-          key = "#{key_type}:#{key_value}"
-          unless report_creds(full_name, key, **credential_opts)
-            vprint_bad("Error when reporting #{full_name} kerberos key #{key}")
+        credential_opts[:type] = :krb_enc_key
+        info[:kerberos_keys].each do |key|
+          krb_enckey = {
+            **key,
+            # Map the SAMR kerberos key to an IANA compatible enctype before persisting
+            enctype: SAMR_KERBEROS_TYPE_TO_IANA[key[:enctype]]
+          }
+
+          krb_enckey_to_s = krb_enc_key_to_s(krb_enckey)
+          key_data = Metasploit::Credential::KrbEncKey.build_data(**krb_enckey)
```

Comment:
Not a blocker, this seems like something that might need to be done consistently with `: krb_enc_key` type creds. Is there a good place to put this in a lib or a helper that could provide a common constant for `SAMR_KERBEROS_TYPE_TO_IANA`?

Reply:
Yeah - I think that's a good idea. It should most likely be in ruby_smb alongside the dcerpc values: https://github.com/rapid7/ruby_smb/blob/278abfa7b8ae9403ad008d1894ac0033460cd51a/lib/ruby_smb/dcerpc/samr.rb#L256-L263

I'll keep that out of scope of this PR for now to avoid the overhead of grabbing in those changes in as part of the feature branch